test/script/tskdbdiff.py

```diff
@@ -292,50 +308,66 @@ def _dump_output_db_nonbb(db_file, dump_file):
-        shutil.copy(db_file, backup_db_file)
-        # We sometimes get situations with messed up permissions
-        os.chmod (backup_db_file, 0o777)
-
-        conn = sqlite3.connect(backup_db_file)
-        id_path_table = build_id_table(conn.cursor())
-        id_vs_parts_table = build_id_vs_parts_table(conn.cursor())
-        id_vs_info_table = build_id_vs_info_table(conn.cursor())
-        id_fs_info_table = build_id_fs_info_table(conn.cursor())
-        id_objects_table = build_id_objects_table(conn.cursor())
-        conn.text_factory = lambda x: x.decode("utf-8", "ignore")
-
-        # Delete the blackboard tables
-        conn.execute("DROP TABLE blackboard_artifacts")
-        conn.execute("DROP TABLE blackboard_attributes")
-
-        # Write to the database dump
-        with codecs.open(dump_file, "wb", "utf_8") as db_log:
-            for line in conn.iterdump():
-                line = normalize_db_entry(line, id_path_table, id_vs_parts_table, id_vs_info_table, id_fs_info_table, id_objects_table)
-                db_log.write('%s\n' % line)
-            # Now sort the file    
+        conn, backup_db_file = db_connect(db_file, isMultiUser, pgSettings)
+        id_path_table = build_id_table(conn.cursor(), isMultiUser)
+        id_vs_parts_table = build_id_vs_parts_table(conn.cursor(), isMultiUser)
+        id_vs_info_table = build_id_vs_info_table(conn.cursor(), isMultiUser)
+        id_fs_info_table = build_id_fs_info_table(conn.cursor(), isMultiUser)
+        id_objects_table = build_id_objects_table(conn.cursor(), isMultiUser)
+
+        if isMultiUser: # Use PostgreSQL
+            os.environ['PGPASSWORD']=pgSettings.password
+            pgDump = ["pg_dump", "--inserts", "-U", pgSettings.username, "-h", pgSettings.pgHost, "-p", pgSettings.pgPort, "-d", db_file, "-E", "utf-8", "-T", "blackboard_artifacts", "-T", "blackboard_attributes", "-f", "postgreSQLDump.sql"]
+            subprocess.call(pgDump)
+            postgreSQL_db = codecs.open("postgreSQLDump.sql", "r", "utf-8")
+            # Write to the database dump
+            with codecs.open(dump_file, "wb", "utf_8") as db_log:
+                dump_line = ''
+                for line in postgreSQL_db:
+                    line = line.strip('\r\n ')
+                    # Deal with pg_dump result file
+                    if line.startswith('--') or line.lower().startswith('alter') or not line: # It's comment or alter statement or empty line
+                        continue
+                    elif not line.endswith(';'): # Statement not finished
+                        dump_line += line
+                        continue
+                    else:
+                        dump_line += line
+                    dump_line = normalize_db_entry(dump_line, id_path_table, id_vs_parts_table, id_vs_info_table, id_fs_info_table, id_objects_table) 
+                    db_log.write('%s\n' % dump_line)
+                    dump_line = ''
+            postgreSQL_db.close()
```

Comment:
Let's be sure to do a test where there is a regression, i.e., the gold is not matched, so that we can verify both the success and failure code paths, thanks!